event/event.go

```diff
@@ -0,0 +1,181 @@
+// event is a package that models events that occur during the execution of a program
+package event
+
+// Event is an interface that custom events should implement.
+// It is empty for now because there are no general methods
+// required of events yet.
+type Event interface {
+}
+
+// KeyCharacter is an event that occurs when a character is actually typed on
+// the keyboard. This may be provided by an input method.
+type KeyCharacter struct {
+	// Code of the key pressed or released
+	Code int
+	// Key board modifiers
+	Modifiers int
+	// Character typed
+	Character rune
+}
+
+// KeyDown is an event that occurs when a key is pressed on the keyboard.
+type KeyDown struct {
+	// Code of the key pressed or released
+	Code int
+	// Key board modifiers
+	Modifiers int
+}
+
+// KeyUp is an event that occurs when a key is released on the keyboard.
+// The data is the same as for a KeyDown event
+type KeyUp KeyDown
+
+// GamepadAxis is for event where an axis on a game pad changes.
+type GamepadAxis struct {
+	// Which gamepad caused the event
+	ID int
+	// Axis of the game pad that changed, if any
+	Axis int
+	// Position of the axis after the change
+	Position float32
+}
+
+//GamepadButtonDown is a game pad button press event.
+type GamepadButtonDown struct {
+	// Which gamepad caused the event
+	ID int
+	// Button that was pressed, if any.
+	Button int
+	// Position of the button after the change.
+	Position float32
+}
+
+//GamepadButtonDown is a game pad button release event.
+//The data is identical to a GamePadButtonDown event.
+type GamepadButtonUp GamepadButtonDown
+
+//GamepadAttach happens when a new game pad is attached.
+type GamepadAttach struct {
+	// Which gamepad caused the event
+	ID int
```

Comment:
Could a gamepad identifier an integer?

Reply:
Sorry I don't quite undserstand your question? In Allegro we used an integer to identify a gamepad. For example, if you get a GamepadAttach event with ID 7, it means that from now on you can expect to revceive other gamepad events with ID also set to 7. Would perhaps you like to use a string ID in stead?

Reply:
NVM, I confirmed identifiers are represented as integers in GLFW and Web gamepad API.